design/dedicated-osd-pod.md

```diff
@@ -0,0 +1,73 @@
+# Run OSD in its own Pod
+
+## TL;DR
+
+A one-OSD-per-Pod placement should be implemented to improve reliability and resource efficiency for Ceph OSD daemon.
+
+## Background
+
+Currently in Rook 0.7, Rook Operator starts a ReplicaSet for Rook OSD  on each storage node. The ReplicaSet has just one replica. Rook OSD scans and prepares devices, creates OSD IDs and data directories, generates Ceph configuration. At last, Rook OSD starts all Ceph OSD daemons in foreground and tracks the OSD processes.
+
+As observed, all Ceph OSDs are running in the same Pod.
+
+## Limitations
+
+One Pod for all OSDs doesn't have the highest reliability nor efficiency. If the Pod is deleted, accidentally or during maintenence, all OSDs are down till the ReplicaSet restart. If nodes have asymmetric configurations, some have more devices than others, the Rook Operator doesn't request appropriate resources for the OSD Pods, since the resources are requested before the OSDs are discovered.
```

Comment:
Could you expand more on the issues with this? Perhaps a bulleted list would help. For example:
- Rook's process management is very simple. Using Kubernetes pod management is much more reliable
- Resource limits cannot be set effectively on the OSDs since the number of osds per in the pod could vary from node to node.
- The operator cannot make decisions about the topology because it doesn't know in advance what devices are available on the nodes.

Reply:
added